### PR TITLE
fix(dashboards): Updates the widget viewer table to also render columns for orderby and groupby

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Updates the widget viewer on `line`, `area`, and `bar` chart widgets with groupbys and limits to:
- display the groupby columns as their own columns in the widget viewer table
- display the orderby as a column if the orderby is not selected as an aggregate or column

image

### CODE REVIEW

**Comment by narsaynorath** [comment]
I was running this locally and realized that the orderby is being added to the widget as an actual field. I was toggling the columns but all you need to do is to open the widget viewer and the orderby gets injected into the state rather than just the request to get the table results


Uploading Screen Recording 2022-05-11 at 4.32.44 PM.mov…

**Comment by narsaynorath** [comment]
https://user-images.githubusercontent.com/22846452/167943572-f9b63e65-c7d7-4903-a705-7ccab5d2aafb.mov


I hit enter too early and my last comment isn't going to let me re-up my file lol

File: static/app/components/modals/widgetViewerModal.tsx [hunk 1]
```diff
@@ -238,11 +243,55 @@ function WidgetViewerModal(props: Props) {
...
+  // that is not explicitly selected as an aggregate or column. We need to explictly
+  // include the orderby in the table widget aggregates and columns otherwise
+  // eventsv2 will complain about sorting on an unselected field.
+  if (
+    widget.widgetType === WidgetType.DISCOVER &&
+    orderby &&
+    !fields.includes(rawOrderby)
+  ) {
+    fields.push(rawOrderby);
+    [tableWidget, primaryWidget].forEach(aggregatesAndColumns => {
```
Comment (narsaynorath) [inline static/app/components/modals/widgetViewerModal.tsx:264]: Is there a particular reason we add it to the `primaryWidget` here? I think this is the reason the widget state gets updated and I don't think it's necessary. The orderby gets handled in `widgetQueries.tsx` if it's not in the fields. 

I like keeping this logic here though

File: static/app/components/modals/widgetViewerModal.tsx [hunk 2]
```diff
@@ -307,17 +359,11 @@ function WidgetViewerModal(props: Props) {
   );
 
   let columnOrder = decodeColumnOrder(
-    tableWidget.queries[0].fields?.map(field => ({
+    fields.map(field => ({
       field,
     })) ?? []
```
Comment (narsaynorath) [inline static/app/components/modals/widgetViewerModal.tsx:364]: I think with us using `fields.map` we can drop the `?? []` since it'll always be an array

**Review by narsaynorath — COMMENTED** [review]
Some small comments and hopefully the `primaryWidget` var will be no problem to remove

File: static/app/components/modals/widgetViewerModal.tsx (hunk 2 above)
Comment (edwardgou-sentry) [reply static/app/components/modals/widgetViewerModal.tsx]: 👍 good observation!

File: static/app/components/modals/widgetViewerModal.tsx [hunk 3]
```diff
@@ -238,11 +243,55 @@ function WidgetViewerModal(props: Props) {
...
+          }
+        });
+      }
+    });
+  }
+
+  // Need to set the orderby of the eventsv2 query to equation[index] format
+  // since eventsv2 does not accept the raw equation as a valid sort payload
+  if (isEquation(rawOrderby) && tableWidget.queries[0].orderby === orderby) {
+    const orderbyEquationIndex = fields.reduce((acc, field) => {
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/components/modals/widgetViewerModal.tsx]: ah nice, this is useful

File: static/app/components/modals/widgetViewerModal.tsx (hunk 1 above)
Comment (narsaynorath) [reply static/app/components/modals/widgetViewerModal.tsx:265]: Yep, makes total sense now that you mention the custom equations. Thanks for clarifying that!

**Review by narsaynorath — APPROVED** [review]
Awesome job! I appreciate the comments you left in the code.

**Pull request merged**